Error reporting

jad-hamza opened the pull request:
* Add an exception handler that hides the stack trace and prints in the file `stainless-debug.txt`
* Use this exception handler in the Stainless Plugin
* Add an option to print the stack trace when there is an error
* Minor error messages changes

romac commented on core/src/main/scala/stainless/MainHelpers.scala:
```diff
@@ -141,81 +142,91 @@ trait MainHelpers extends inox.MainHelpers { self =>
+      setup(args)
+    } catch {
+      case _: Throwable =>
+        println("There was an exception while parsing command-line options")
```
This should probably print to stderr instead of stdout.

romac replied:
Or perhaps just use `sys.error` instead of `println` + `System.exit`.

romac commented on frontends/dotty/src/main/scala/stainless/frontends/dotc/CodeExtraction.scala:
```diff
@@ -279,7 +279,7 @@ class CodeExtraction(inoxCtx: inox.Context, cache: SymbolsContext)(implicit val
 
       case other =>
-        reporter.warning(other.pos, "Could not extract tree in static container: " + other)
+        reporter.warning(other.pos, s"Stainless does not support `$other` in static containers")
```
If `$other` is a large tree then the output might not be very readable since the error message will be split in two by a tree potentially spanning multiple lines. Might thus be better to keep the offending tree at the end of the error message.

romac commented on frontends/dotty/src/main/scala/stainless/frontends/dotc/CodeExtraction.scala:
```diff
@@ -510,7 +510,7 @@ class CodeExtraction(inoxCtx: inox.Context, cache: SymbolsContext)(implicit val
 
       case other =>
-        reporter.warning(other.pos, "Could not extract tree in class: " + other)
+        reporter.warning(other.pos, s"Stainless does not support `$other` in class $id")
```
Same here.

romac commented on frontends/scalac/src/main/scala/stainless/frontends/scalac/CodeExtraction.scala:
```diff
@@ -297,7 +297,7 @@ trait CodeExtraction extends ASTExtractors {
 
       case other =>
-        reporter.warning(other.pos, "Could not extract tree in static container: " + other)
+        reporter.warning(other.pos, s"Stainless does not support `$other` in static containers")
```
Same here.

romac commented on frontends/scalac/src/main/scala/stainless/frontends/scalac/CodeExtraction.scala:
```diff
@@ -474,7 +474,7 @@ trait CodeExtraction extends ASTExtractors {
 
       case other =>
-        reporter.warning(other.pos, "Could not extract tree in class: " + other + " (" + other.getClass + ")")
+        reporter.warning(other.pos, s"Stainless does not support `$other` in class $id")
```
Same here.

jad-hamza commented:
Thanks for the feedback! I also removed `regularRunCycle` (which was calling `internalError` and displaying stack traces), and I added a call to the error handler for options parsing exceptions
